plugins/parsers/csv/README.md

```diff
@@ -33,8 +33,35 @@ values.
   ## If this is not specified, type conversion will be done on the types above.
   csv_column_types = []
 
-  ## Indicates the number of rows to skip before looking for header information.
+  ## Indicates the number of rows to skip before looking for metadata and header information.
   csv_skip_rows = 0
+  
+  
+  ## Indicates the number of rows to parse as metadata before looking for header information. 
+  ## By default, the parser assumes there are no metadata rows to parse. 
+  ## If set, the parser would use the provided separators in the cvs_metadata_separators to look for metadata.
+  csv_metadata_rows = 0
+  
+  ## A list of metadata separators. If csv_metadata_rows is set,
+  ## cvs_metadata_separators must contain at least one separator.
+  ## Please note that separators are case sensitive.
+  ## For example:
+  ## cvs_metadata_separators = [",", ":", "="]
+  ## to parse "Key1,Value1" "Key2:Value2" "Key3=Value3"
+  ## the returned (key,value) pares would be as follows:
+  ## (Key1,Value1), (Key2,Value2), (Key3,Value3)
+  cvs_metadata_separators = [","]
+  
+  ## A set of metadata trim characters. 
+  ## If cvs_metadata_trim_cutset is not set, no trimming is performed.
+  ## Please note that the trim cutset is case sensitive.
+  ## For example, given the following:
+  ## cvs_metadata_separators = [",", ":", "="]
+  ## cvs_metadata_trim_set = " #'"
+  ## to parse "# Version = 1.1" "Creation Date: '2021-11-01'"
+  ## the returned (key,value) pares would be as follows:
+  ## (Version,1.1), (Creation Date,2021-11-01)
+  cvs_metadata_trim_set = ""
```

Comment:
`separators` is an array and `trim_set` is a string, that doesn't seem consistent..

Reply:
When trimming you give a string of characters, but when splitting key value pairs you need to have a unique set of separator strings ordered by their length.

Reply:
@srebhan what are your thoughts on this?

Reply:
I tend to agree with @etycomputer as this is a set of characters. Maybe we should name the option `csv_metadata_trim_characters`?